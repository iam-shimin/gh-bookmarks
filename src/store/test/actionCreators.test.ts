import {
	addBookmark,
	removeBookmark,
	renameBookmark
} from '../actionCreators';

describe('ActionCreators', () => {
	test('add bookmark', () => {
		const user = {
			login: 'login',
			id: 'id',
			avatar_url: 'avatar_url',
			html_url: 'html_url'
		};

		const action = addBookmark({
				id: 'id',
				name: 'name',
				html_url: 'html_url',
				owner: user,
				full_name: 'full_name',
				displayName: 'displayName',
				description: 'description'
		});

		expect(action).toMatchSnapshot();
	});

	test('remove Bookmark', () => {
		const action = removeBookmark('id');
		expect(action).toMatchSnapshot();
	});

	test('rename bookmark', () => {
		const action = renameBookmark('id', 'newName');
		expect(action).toMatchSnapshot();
	})
})