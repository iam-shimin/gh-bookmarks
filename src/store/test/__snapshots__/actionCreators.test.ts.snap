// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionCreators add bookmark 1`] = `
Object {
  "payload": Object {
    "description": "description",
    "displayName": "displayName",
    "full_name": "full_name",
    "html_url": "html_url",
    "id": "id",
    "name": "name",
    "owner": Object {
      "avatar_url": "avatar_url",
      "html_url": "html_url",
      "id": "id",
      "login": "login",
    },
  },
  "type": "BOOKMARK_ADD",
}
`;

exports[`ActionCreators remove Bookmark 1`] = `
Object {
  "payload": Object {
    "id": "id",
  },
  "type": "BOOKMARK_REMOVE",
}
`;

exports[`ActionCreators rename bookmark 1`] = `
Object {
  "payload": Object {
    "id": "id",
    "newName": "newName",
  },
  "type": "BOOKMARK_RENAME",
}
`;
